File: Python/functions/no checking/harshad.py
#harshad no
def harshad(n):
    s=0
    for i in str(n):
        s+=int(i)
    if n%s==0:
        return True
    else:
        return False
n=int(input("Enter hte no. = "))
if harshad(n):
    print("Harshad no.")
else:
    print("not Harshad no.")

File: C Programms/No. Cheking In Range/harshad.c
//WAp to print Harshad no.  from 1 - n
#include<stdio.h>
void main()
{
    int n,r,s,x;
    printf("Enter the range = ");
    scanf("%d",&r);
    printf("Harshad No. from 1 to %d are ...\n",r);
    for(n=1;n<=r;n++)
    {
        for(x=n,s=0;x!=0;x/=10)
            s+=x%10;
        if(n%s==0)
            printf("%d ",n);
    }
}